From 536155d650540951b88bb49b13571d709a57ad4d Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Sat, 28 Aug 2021 15:16:56 -0700
Subject: [PATCH 136/196] Make XAddress parser C+11 compliant (#216)

The constructor used in part of the implementation for XAddresses is
using a `std::vector` constructor that is not supported until C++20. This
commit instead uses `std::copy` to perform the same operation.
---
 src/addresses/xaddress.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/addresses/xaddress.cpp b/src/addresses/xaddress.cpp
index 27dbd8b53..60b9bb657 100644
--- a/src/addresses/xaddress.cpp
+++ b/src/addresses/xaddress.cpp
@@ -16,6 +16,7 @@
 
 #include <assert.h>
 #include <string>
+#include<algorithm>
 
 namespace XAddress {
 /**
@@ -95,8 +96,9 @@ DecodeError Decode(const std::string &address, Content &parsedOutput) {
         return UNDERSIZED_PAYLOAD;
     }
     const AddressType addressByte = AddressType(vch[0]);
-    parsedOutput = Content(token, networkByte, addressByte,
-                           std::vector(vch.begin() + 1, vch.end() - 4));
+    std::vector<uint8_t> payload(vch.size() - 5);
+    std::copy(vch.begin() + 1, vch.end() - 4, payload.begin());
+    parsedOutput = Content(token, networkByte, addressByte, payload);
     const uint256 check = HashAddressContents(parsedOutput);
     if (!memcmp(&check, &vch[vch.size() - 4], 4)) {
         return DECODE_OK;
-- 
2.39.5 (Apple Git-154)

